### PR TITLE
[CEDS-3296] Supplementary Backend data model update

### PR DESCRIPTION
**Pull request by glidester**
Add new mrn and eidrDateStamp fields to the
consignment references model entity.

### CODE REVIEW

File: test/unit/uk/gov/hmrc/exports/services/mapping/goodsshipment/PreviousDocumentsBuilderSpec.scala [hunk 1]
```diff
@@ -19,18 +19,83 @@ package uk.gov.hmrc.exports.services.mapping.goodsshipment
...
 import wco.datamodel.wco.dec_dms._2.Declaration.GoodsShipment
 
 class PreviousDocumentsBuilderSpec extends UnitSpec with ExportsDeclarationBuilder {
 
   "PreviousDocumentsBuilder " should {
 
     "correctly map new model to a WCO-DEC GoodsShipment.PreviousDocuments instance" when {
-      "when document data is present and a DUCR has been added previously" in {
 
+      Seq(STANDARD, SIMPLIFIED, OCCASIONAL, CLEARANCE).foreach { decType =>
```
Comment (MaciejRewera) [inline test/unit/uk/gov/hmrc/exports/services/mapping/goodsshipment/PreviousDocumentsBuilderSpec.scala:31]: What about Supplementary dec?

File: test/unit/uk/gov/hmrc/exports/services/mapping/goodsshipment/PreviousDocumentsBuilderSpec.scala [hunk 2]
```diff
@@ -19,18 +19,83 @@ package uk.gov.hmrc.exports.services.mapping.goodsshipment
...
+          val previousDocs = goodsShipment.getPreviousDocument
+          previousDocs.size must be(1)
+          previousDocs.get(0).getID.getValue must be(UCRBuilderSpec.correctConsignmentReferences.ducr.ducr)
+          previousDocs.get(0).getCategoryCode.getValue must be("Z")
+          previousDocs.get(0).getTypeCode.getValue must be("DCR")
+          previousDocs.get(0).getLineNumeric must be(BigDecimal(1).bigDecimal)
+        }
+      }
+
+      s"a DUCR is specified for a SUPPLEMENTARY declaration" in {
```
Comment (MaciejRewera) [inline test/unit/uk/gov/hmrc/exports/services/mapping/goodsshipment/PreviousDocumentsBuilderSpec.scala:46]: Nit: No need for 's' before quotes.

**Review by MaciejRewera — APPROVED** [review]
LGTM

File: test/unit/uk/gov/hmrc/exports/services/mapping/goodsshipment/PreviousDocumentsBuilderSpec.scala (hunk 2 above)
Comment (udinobi) [inline test/unit/uk/gov/hmrc/exports/services/mapping/goodsshipment/PreviousDocumentsBuilderSpec.scala:46]: Nit: not an interpolated string. The `s` can be removed.

**Pull request merged**